src/cp_parallel/cp_parallel.py

```diff
@@ -0,0 +1,153 @@
+    """
+    # Access the command (args) and stderr (output) for each CompletedProcess object
+    for result in results:
+        # Extract plate name and decode output
+        plate_name = result.args[6].name
+        output_string = result.stderr.decode("utf-8")
+        log_file_path = log_dir / f"{plate_name}_{run_name}_run.log"
+
+        # Create and configure a logger for this plate
+        logger = logging.getLogger(f"{run_name}_{plate_name}")
+        logger.setLevel(logging.INFO)
+
+        # Avoid adding multiple handlers to the same logger
+        if not logger.handlers:
+            fh = FileHandler(log_file_path)
+            log_format = "[%(asctime)s] [Process ID: %(process)d] %(message)s"
+            formatter = Formatter(log_format)
+            fh.setFormatter(formatter)
+            logger.addHandler(fh)
+
+        # Log the information
+        logger.info(f"Plate Name: {plate_name}")
+        logger.info(f"Output String: {output_string}")
+
+        # Explicitly close and remove the handler after logging
+        for handler in logger.handlers[:]:
+            handler.close()
+            logger.removeHandler(handler)
+
+
+def run_cellprofiler_parallel(
+    plate_info_dictionary: dict,
+    run_name: str,
+) -> None:
+    """
+    This function utilizes multi-processing to run CellProfiler pipelines in parallel.
+
+    Args:
+        plate_info_dictionary (dict): dictionary with all paths for CellProfiler to run a pipeline
+        run_name (str): a given name for the type of CellProfiler run being done on the plates (example: whole image features)
+
+    Raises:
+        FileNotFoundError: if paths to pipeline and images do not exist
+    """
+    # create a list of commands for each plate with their respective log file
+    commands = []
+
+    # make logs directory
+    log_dir = pathlib.Path("./logs")
+    os.makedirs(log_dir, exist_ok=True)
+
+    # iterate through each plate in the dictionary
+    for _, info in plate_info_dictionary.items():
+        # set paths for CellProfiler
+        path_to_pipeline = info["path_to_pipeline"]
+        path_to_images = info["path_to_images"]
+        path_to_output = info["path_to_output"]
+
+        # check to make sure paths to pipeline and directory of images are correct before running the pipeline
+        if not pathlib.Path(path_to_pipeline).resolve(strict=True):
+            raise FileNotFoundError(
+                f"The file '{pathlib.Path(path_to_pipeline).name}' does not exist"
+            )
+        if not pathlib.Path(path_to_images).is_dir():
+            raise FileNotFoundError(
+                f"Directory '{pathlib.Path(path_to_images).name}' does not exist or is not a directory"
+            )
+        # make output directory if it is not already created
+        pathlib.Path(path_to_output).mkdir(exist_ok=True)
+
+        # creates a command for each plate in the list
+        command = [
+            "cellprofiler",
```

Comment:
I think this can be debated. I personally only see this function as a utility to someone who is already installing CellProfiler in their own environment. I do not personally see this as software to have to include it as a dependency.

@MikeLippincott feel free to include your feedback here.

Reply:
Could the CellProfiler CLI be used through the Python package somehow (maybe this is future issue / PR work)? Documenting where CellProfiler should come from could be a path to success in the meantime. Consider if someone knows nothing about CellProfiler and was told to use this project to get started - what would they need to know and how could they know it?

Reply:
I have added this as an issue, thank you for bringing this up!